### DISCUSSION TITLE
Connection timed out

### DISCUSSION THREAD

**Discussion by danielmanser** [antonym/netboot.xyz#921] [General] [answered]
I'm trying to use netboot.xyz via the [PXE boot option](https://netboot.xyz/booting/tftp/) and managed to configure the boot file `netboot.xyz.kpxe`, which seems to work. At least the client is loading the file:

```
Intel UNDI, PXE-2.0 (build 082)
Copyright (C) 1997,1998,1999  Intel Corporation
VIA Rhine III Management Adapter v2.43 (2005/12/15)

CLIENT MAC ADDR: 00 0D B9 18 8A 38
CLIENT IP: 10.0.0.68  MASK: 255.255.255.0  DHCP IP: 10.0.0.1
GATEWAY IP: 10.0.0.1
PXE->EB: !PXE at 9E2E:0070, entry point at 9E2E:0106
         UNDI code segment 9E2E:1966, data segment 94BD:9710 (594-640kB)
         UNDI device is PCI 00:09.0, type DIX+802.3
         594kB free base memory after PXE unload
iPXE initialising devices...ok



iPXE 1.21.1+ (g3c04) -- Open Source Network Boot Firmware -- http://ipxe.org
Features: DNS HTTP HTTPS iSCSI TFTP VLAN AoE ELF MBOOT PXE bzImage COMBOOT Menu PXEXT
netboot.xyz - v2.x

Configuring (net0 00:0d:b9:18:8a:38)...... ok
tftp://10.0.0.1/local-vars.ipxe... No such file or directory (http://ipxe.org/2d12603b)
Local vars file not found... attempting TFTP boot...
tftp://10.0.0.1/HOSTNAME-alixoffice.ipxe... No such file or directory (http://ipxe.org/2d12603b)
Custom boot by Hostname not found trying MAC...
tftp://10.0.0.1/MAC-000db9188a38.ipxe... No such file or directory (http://ipxe.org/2d12603b)
Custom boot by MAC not found booting default...
tftp://10.0.0.1/menu.ipxe... No such file or directory (http://ipxe.org/2d12603b)
Local TFTP failed... attempting remote HTTPS
Could not start download: Invalid argument (http://ipxe.org/1c25e002)
HTTPS appears to have failed... attempting HTTP
```

However. the connection to http://boot.netboot.xyz/menu.ipxe is timing out:

```
http://boot.netboot.xyz/menu.ipxe................... Connection timed out (http://ipxe.org/4c0a6035)
HTTP has failed, localbooting...
```

If using a client in the same subnet, I can sucessfully download the file. Am I doing something wrong?

To clarify things: I've tried both the `netboot.xyz.kpxe` and `netboot.xyz-undionly.kpxe` files, with the same result.

**Comment by antonym** [comment, marked as answer]
If you load up the command line, run DHCP, and then run route, what does it show?  Sometimes it'll ping up v6 addresses and attempt to use those even though you are using v4.

**Reply by danielmanser** [reply]
For some reason the subnet mask wasn't filled in. I've tried with another computer on my network, same issue. I guess it's something with my network then. Thank you 👍

**Reply by logclinker** [reply]
I'm having the same issue, but I can not figure out what exactly has fixed the problem for you, and where exactly where you missing the subnet mask?

**Comment by dan3805** [comment]
I had the same problem. To solve it, I opened the iPXE shell and entered the command `set dns 1.1.1.1` After that, I was able to continue. 🛠️👍

**Reply by logclinker** [reply]
I tried this. it didn't do anything for me.

**Reply by xandaaah** [reply]
This fixed it for me.
